### PR TITLE
Replace tournament context variables with new variables

### PR DESCRIPTION
**Pull request by lucaasrojas**
- Remove Ongoing Tournament
- Add Costs and Additionals context variables

### CODE REVIEW

File: demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/Models/UserContext.cs [hunk 1]
```diff
@@ -8,7 +8,9 @@ public class UserContext
 
         public bool UseTextAnalytics { get; set; }
```
Comment (ParadoxARG) [inline demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/Models/UserContext.cs]: We should remove Referrer and UseTextAnalytics.

File: demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/wwwroot/js/demo.js [hunk 2]
```diff
@@ -267,12 +270,6 @@ function setupContextControls() {
...
 
-    const currentTournamentSelectEle = document.getElementById('currentTournament');
-    currentTournamentSelectEle.selectedIndex = ramdomizeSelectedOption(currentTournamentSelectEle);
-    currentTournamentSelectEle.addEventListener('change', (event) => {
-        updateContext(null, event.target.value);
-    });
-
     const deviceSelectEle = document.getElementById('device');
     deviceSelectEle.selectedIndex = ramdomizeSelectedOption(deviceSelectEle);
     deviceSelectEle.addEventListener('change', (event) => {
```
Comment (ParadoxARG) [inline demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/wwwroot/js/demo.js:264]: Remove the second `null`(tournament) from updateContext in `deviceSelectEle` listener and `UseUserAgentEle` listener

**Review by ParadoxARG — CHANGES_REQUESTED** [review]

File: demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/Controllers/MetadataController.cs [hunk 3]
```diff
@@ -17,7 +17,7 @@ public MetadataController(IActionsRepository actionsRepository)
         [HttpGet("Actions")]
         public JsonResult Actions(bool useTextAnalytics)
```
Comment (ParadoxARG) [inline demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/Controllers/MetadataController.cs]: Remove parameter

File: demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/Repositories/ActionsRepository.cs [hunk 4]
```diff
@@ -18,14 +18,14 @@ public ActionsRepository(IArticleRepository articleRepository)
...
-            return useTextAnalytics ? _actionsWithTextAnalytics.Cast<RankableAction>().ToList() : _actions.Cast<RankableAction>().ToList();
+            return _actions.Cast<RankableAction>().ToList();
         }
 
-        public IList<RankableActionWithMetadata> GetActionsWithMetadata(bool useTextAnalytics)
+        public IList<RankableActionWithMetadata> GetActionsWithMetadata()
         {
-            return useTextAnalytics ? _actionsWithTextAnalytics : _actions;
+            return _actions;
         }
```
Comment (ParadoxARG) [inline demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/Repositories/ActionsRepository.cs:29]: remove all the _ActionsWithTextTanalytics logic

File: demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/wwwroot/js/demo.js [hunk 5]
```diff
@@ -220,8 +222,9 @@ document.addEventListener("DOMContentLoaded", function () {
 
 let context = {
     referrer: "social",
-    tournament: "wimbledon",
     device: "mobile",
+    packageAdditionals: getRandomOption(additonalsOptions),
+    costs: getRandomOption(costsOptions),
     userAgent: null,
     useTextAnalytics: false
 };
```
Comment (ParadoxARG) [inline demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/wwwroot/js/demo.js]: Remove referrers and actions listeners and logic

**Review by ParadoxARG — CHANGES_REQUESTED** [review]

File: demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/wwwroot/js/demo.js [hunk 6]
```diff
@@ -250,65 +251,46 @@ function clearRewardmessage() {
...
     });
 
-    updateContext(referrerSelectEle.value, currentTournamentSelectEle.value, deviceSelectEle.value);
+    updateContext(deviceSelectEle.value);
 }
 
-function updateContext(referrer, currentTournament, device, removeUserAgent, userAgent) {
-    context.referrer = referrer || context.referrer;
-    context.tournament = currentTournament || context.tournament;
+function updateContext( device, currentCost, currentAdditionals, removeUserAgent, userAgent) {
```
Comment (ParadoxARG) [inline demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/wwwroot/js/demo.js]: Make all the calls consistent with the new firm

**Review by ParadoxARG — CHANGES_REQUESTED** [review]

File: demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/wwwroot/js/demo.js [hunk 7]
```diff
@@ -250,65 +251,46 @@ function clearRewardmessage() {
...
-        updateContext(null, null, event.target.value);
+        updateContext(event.target.value, null, null, false, null);
     });
 
     const UseUserAgentEle = document.getElementById('use-useragent');
     UseUserAgentEle.addEventListener('change', (event) => {
         const checkbox = event.target;
         if (checkbox.checked) {
-            updateContext(null, null, null, false, userAgent);
+            updateContext(null, null, false, userAgent);
```
Comment (ParadoxARG) [inline demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/wwwroot/js/demo.js]: missing a null parameter

File: demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/wwwroot/js/demo.js [hunk 8]
```diff
@@ -250,65 +251,46 @@ function clearRewardmessage() {
...
         } else {
-            updateContext(null, null, null, true);
+            updateContext(null, null, true, null);
```
Comment (ParadoxARG) [inline demos/PersonalizerTravelAgencyDemo/PersonalizerTravelAgencyDemo/wwwroot/js/demo.js]: Missing a null parameter

**Review by ParadoxARG — CHANGES_REQUESTED** [review]

**Review by ParadoxARG — APPROVED** [review]
LGTM

**Review by kbrons — APPROVED** [review]

**Pull request merged**